src/main/java/org/truffleruby/parser/parser/RubyParser.y

```diff
@@ -1383,6 +1383,13 @@ command_args    : /* none */ {
 block_arg       : tAMPER arg_value {
                     $$ = new BlockPassParseNode(support.getPosition($2), $2);
                 }
+		| tAMPER {
+		    if (!support.local_id(ParserSupport.FORWARD_ARGS_BLOCK_VAR)) {
+		        support.yyerror("no anonymous block parameter");
+		    }
+
+		    $$ = new BlockPassParseNode(lexer.tokline, new LocalVarParseNode(support.getPosition(null), 0, ParserSupport.FORWARD_ARGS_BLOCK_VAR));
```

Comment:
Could you add `FORWARD_ARGS_BLOCK_VAR_TSTRING`?
Then we don't need `shadowing_lvar(String)`.

Reply:
Will do, but I did like having it close to the other forwarding code. Our implementation of `...` forwarding is quite different than MRI's, so tracing how this all works was not very straightforward.

Reply:
Okay, taking another look, I misunderstood what you were suggesting. I thought you wanted me to put it in `TStringConstants`, but it looks you meant treat it more like `FORWARD_ARGS_KWREST_VAR_TSTRING`. In that case, I don't see the harm in having the j.l.String method. It adds a one line method body that doesn't require the awkward boxing and unboxing of TruffleString to j.l.String and subsequent interning.

Reply:
I've amended the original commit.